core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -104,9 +104,11 @@ public String newDataLocation(String filename) {
   }
 
   static class ObjectStoreLocationProvider implements LocationProvider {
-    private static final Function<Object, Integer> HASH_FUNC =
-        Transforms.bucket(Integer.MAX_VALUE).bind(Types.StringType.get());
 
+    private static final HashFunction HASH_FUNC = Hashing.sha1();
```

Comment:
Thought about this a bit more. There is a performance implication of using SHA1. Can we continue to use MurMur3 which is fast? It can produce a 128bit hash, and we can convert that to a base64 URL safe string using `Base64.getUrlEncoder().encodeToString(hash)`

Reply:
+1 for Murmur3. No need for SHA1.